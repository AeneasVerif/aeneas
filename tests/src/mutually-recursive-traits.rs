//@ [lean] known-failure
//@ [!lean] skip
pub trait Trait1 {
    type T: Trait2;
}

pub trait Trait2: Trait1 {}

pub trait T1<T: T2<Self>>: Sized {}
pub trait T2<T: T1<Self>>: Sized {}
