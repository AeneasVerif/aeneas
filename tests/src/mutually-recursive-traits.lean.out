[[92mInfo[39m ] Imported: tests/llbc/mutually_recursive_traits.llbc
[[91mError[39m] In file Translate.ml, line 813:
Mutually recursive trait declarations are not supported

Uncaught exception:
  
  (Failure
    "In file Translate.ml, line 813:\
   \nIn file Translate.ml, line 813:\
   \nMutually recursive trait declarations are not supported")

Raised at Aeneas__Errors.craise_opt_span in file "Errors.ml", line 47, characters 4-76
Called from Stdlib__List.iter in file "list.ml", line 110, characters 12-15
Called from Aeneas__Translate.extract_definitions in file "Translate.ml", line 839, characters 2-52
Called from Aeneas__Translate.extract_file in file "Translate.ml", line 966, characters 2-36
Called from Aeneas__Translate.translate_crate in file "Translate.ml", line 1515, characters 5-42
Called from Dune__exe__Main in file "Main.ml", line 317, characters 14-66
